File: get_urls.js
const https = require('https');

function fetchOgImage(url) {
    return new Promise((resolve, reject) => {
        https.get(url, (res) => {
            let data = '';
            res.on('data', (chunk) => data += chunk);
            res.on('end', () => {
                const match = data.match(/<meta\s+property="og:image"\s+content="([^"]+)"/);
                if (match) resolve(match[1]);
                else resolve('Not found');
            });
        }).on('error', reject);
    });
}

async function run() {
    console.log("Daily Finance:", await fetchOgImage('https://play.google.com/store/apps/details?id=com.probeshpath.dailyfinance'));
    console.log("HRM:", await fetchOgImage('https://play.google.com/store/apps/details?id=com.logicsoftbd.hrm'));
    console.log("Smart Track:", await fetchOgImage('https://apps.apple.com/gb/app/platform-approvals/id6759209301'));
}

run();

File: get_urls.py
import urllib.request
import re

def get_og_image(url):
    try:
        req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
        html = urllib.request.urlopen(req).read().decode('utf-8')
        match = re.search(r'<meta property="og:image" content="(.*?)"', html)
        if match:
            return match.group(1)
        
        match2 = re.search(r'<meta property="og:image" content=\'(.*?)\'', html)
        if match2:
            return match2.group(1)
            
        match3 = re.search(r'<meta name="twitter:image" content="(.*?)"', html)
        if match3:
            return match3.group(1)
            
        return "Not found"
    except Exception as e:
        return str(e)

print("Daily Finance:", get_og_image('https://play.google.com/store/apps/details?id=com.probeshpath.dailyfinance'))
print("HRM:", get_og_image('https://play.google.com/store/apps/details?id=com.logicsoftbd.hrm'))
print("Smart Track:", get_og_image('https://apps.apple.com/gb/app/platform-approvals/id6759209301'))
